scripts/fix_unclaimed_records.py

```diff
@@ -0,0 +1,35 @@
+import sys
+import logging
+from website.app import setup_django
+
+setup_django()
+from osf.models import OSFUser
+logger = logging.getLogger(__name__)
+
+
+def add_missing_unclaimed_record(user, node, dry_run):
+    # Get referrer from logs
+    for log in node.logs.filter(action='contributor_added').order_by('date'):
+        if user._id in log.params['contributors']:
+            if log.user_id:
+                referrer = OSFUser.objects.get(id=log.user_id)
+                user.add_unclaimed_record(node, referrer, user.fullname)
+                if not dry_run:
+                    user.save()
+                logger.info('User {} has been given an unclaimed record with name {} for node {}'.format(user._id, user.fullname, node._id))
+            else:
+                logger.info('User {} has could not be given a record, because their referer was anonymous'.format(user._id))
+
+
+def main(dry_run=True):
+    users = OSFUser.objects.filter(date_disabled=None,
+                                   is_registered=False).filter(nodes__type='osf.node').include(None).distinct()
```

Comment:
Also, if any user has registered a preprint with an unregistered contributor made this way, then registrations need to be cleaned up too. 

I had Matt run this quick:
```python
In [1]: %paste
users = OSFUser.objects.filter(date_disabled=None, is_registered=False).filter(nodes__type='osf.registration').include(None).distinct()
for user in users:
    for node in user.nodes.filter(type='osf.registration'):
        if user.unclaimed_records.get(node._id) is None:
            print "At least 1 bad registration"
            assert False

## -- End pasted text --
At least 1 bad registration
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-1-c17d81e0936a> in <module>()
      4         if user.unclaimed_records.get(node._id) is None:
      5             print "At least 1 bad registration"
----> 6             assert False

AssertionError:
```

So there is at least one bad registration